programs/mmm/src/instructions/mip1/sol_mip1_fulfill_buy.rs

```diff
@@ -205,6 +214,44 @@ pub fn handler<'info>(
     )
     .map_err(|_| MMMErrorCode::NumericOverflow)?;
 
+    // check creator_accounts and verify the remaining accounts
+    let creator_accounts = if pool.using_shared_escrow() {
```

Comment:
can we abstract this into a util function?

Reply:
I thought about it, but the function's input array is going to long, so it doesn't actually save that much in terms of reusability, as it's a rewrite of input params

Reply:
refactored a bit more